bytecode/instr.py

```diff
@@ -282,6 +314,33 @@ def stack_effect(self, jump=None):
         else:
             return dis.stack_effect(self._opcode, arg, jump=jump)
 
+    def pre_and_post_stack_effect(self, jump=None, check_pre_and_post=True):
+        _effect = self.stack_effect(jump=jump)
+
+        if not check_pre_and_post:
+            return _effect, 0
+
+        # To compute pre size and post size to avoid segfault cause by not enough stack element
+        _opname = _opcode.opname[self._opcode]
+        if _opname.startswith("DUP_TOP"):
+            return _effect * -1, _effect * 2
+        if _pushes_back(_opname):
+            # if the op pushes value back to the stack, then the stack effect given by dis.stack_effect
+            # actually equals pre + post effect, therefore we need -1 from the stack effect as a pre
+            # condition
+            return _effect - 1, 1
+        if _opname.startswith("UNPACK_"):
+            # Instr(UNPACK_* , n) pops 1 and pushes n
+            # _effect = n - 1
+            # hence we return -1, _effect + 1
+            return -1, _effect + 1
+        if _opname == "FOR_ITER" and not jump:
```

Comment:
Having a test for this case would be nice

Reply:
I think I'd like some help on constructing a test case for this one, it would be great if you can help :)